File: plugin/eval/fixtures/go-patterns/tree/ports_test.go
package ports

import "testing"

func TestParsePort(t *testing.T) {
	cases := []struct {
		name    string
		in      string
		want    int
		wantErr bool
	}{
		{"registered port", "8080", 8080, false},
		{"garbage", "http", 0, true},
		{"out of range", "70000", 0, true},
	}
	for _, tc := range cases {
		t.Run(tc.name, func(t *testing.T) {
			got, err := ParsePort(tc.in)
			if (err != nil) != tc.wantErr {
				t.Fatalf("ParsePort(%q) error = %v, wantErr %v", tc.in, err, tc.wantErr)
			}
			if got != tc.want {
				t.Fatalf("ParsePort(%q) = %d, want %d", tc.in, got, tc.want)
			}
		})
	}
}

File: plugin/eval/fixtures/typescript-patterns/tree/src/ports.test.ts
import { describe, expect, it } from "vitest";

import { parsePort } from "./ports";

describe("parsePort", () => {
  it("accepts a registered port", () => {
    expect(parsePort("8080")).toBe(8080);
  });

  it("rejects garbage and out-of-range values", () => {
    expect(() => parsePort("http")).toThrow(RangeError);
    expect(() => parsePort("70000")).toThrow(RangeError);
  });
});
